packages/graphql-elasticsearch-transformer/src/SearchableModelTransformer.ts

```diff
@@ -225,4 +233,18 @@ export class SearchableModelTransformer extends Transformer {
     const primaryKeySchemaElement = tableResource.Properties.KeySchema.find((keyElement: any) => keyElement.KeyType === 'HASH');
     return primaryKeySchemaElement.AttributeName;
   }
+
+  private getStreamingFunctionPath = (isProjectUsingDataStore: boolean) => {
+    const streamingFuncZipName = isProjectUsingDataStore ? 'streaming-lambda-external-version.zip' : 'streaming-lambda.zip';
+    return path.resolve(path.join(__dirname, '..', 'lib', streamingFuncZipName));
+  };
+
+  private isProjectUsingDataStore = (resolverConfig?: ResolverConfig) => {
```

Comment:
We seem to need this logic in other categories now (e.g. determining if it's a sync enabled endpoint) such as the API category. We're actually considering adding an update to the `aws-exports.js` and `amplifyconfiguration.json` under the API section with something like `syncEnabled:true` and have the libraries give better configuration time warnings/recovery to customers who seem to be incorrectly configuring DataStore or using sync enabled endpoints incorrectly with API category (hence all the issues with "how do I know what `_version` to send?").

So food for thought - Can we pull this detection out higher into the CLI now to use it in multiple places rather than just the Searchable Transformer?

Reply:
Will pull it up into the TransformerContext in graphql-transformer-core